frontend/src/metabase/entities/databases.js

```diff
@@ -41,11 +47,32 @@ const Databases = createEntity({
           reload,
         }),
     ),
+
+    fetchIdfields: createThunkAction(
+      FETCH_DATABASE_IDFIELDS,
+      ({ id }) => async () => {
+        const idfields = await MetabaseApi.db_idfields({ dbId: id });
+        const idfieldsWithDisplayName = idfields.map(field => {
+          field.displayName =
+            field.table.display_name + " → " + field.display_name;
+          return field;
+        });
+        return normalize(idfieldsWithDisplayName, [Fields.schema]);
+      },
+    ),
   },
 
   selectors: {
     getHasSampleDataset: state =>
       _.any(Databases.selectors.getList(state), db => db.is_sample),
+    getIdfields: (state, databaseId) =>
+      Object.values(getMetadata(state).fields)
+        .filter(f => f.table.db_id === databaseId && f.isPK())
+        .map(field => {
+          field.displayName =
+            field.table.display_name + " → " + field.display_name;
+          return field;
+        }),
```

Comment:
I copied over `fetchIdfields` from elsewhere, but then ended up duplicating the `displayName` logic in `getIdFields`. This was done so that the FieldApp can use the idfields that it has already fetched.

This is a bit awkward, so let me know if there are any conventions on what to do here. If we're mutating data after an api call, is it better to do that in a selector or reducer?